packages/std/src/ado_contract/execute.rs

```diff
@@ -74,6 +75,14 @@ impl<'a> ADOContract<'a> {
                 AndromedaMsg::PermissionAction { action } => {
                     self.execute_permission_action(ctx, action)
                 }
+                #[cfg(feature = "rates")]
+                AndromedaMsg::Rates(rates_message) => match rates_message {
+                    RatesMessage::SetRate { action, rate } => {
```

Comment:
Abstract this in to `rates` module under `execute_rates`. Allows for simpler management in future.